### ISSUE TITLE
MPC-HC Inno Installer

### ISSUE THREAD

**Issue by olympus123** [clsid2/mpc-hc#1568]
Hi, thank you for this wonderful program.
There are issues with the MPC-HC Inno installer.
If somebody associates various extensions with MPC-HC from Options -> Player -> Formats, the installer doesn't delete the associations and the registration of MPC-HC in windows during uninstallation.
All these registry keys MPC-HC creates should be deleted during uninstallation.
The only way to remove them is manually, but the uninstaller should do that.
This is what I delete to avoid having leftovers after uninstallation.
The best would be for uninstaller to run the /unregall command before starting uninstallation.

REG DELETE "HKLM\Software\RegisteredApplications" /v "Media Player Classic" /f
REG DELETE "HKLM\Software\Clients\Media\Media Player Classic" /f

REG DELETE "HKCR\mplayerc64.3g2" /f
REG DELETE "HKCR\mplayerc64.3ga" /f
REG DELETE "HKCR\mplayerc64.3gp" /f
REG DELETE "HKCR\mplayerc64.3gp2" /f
REG DELETE "HKCR\mplayerc64.3gpp" /f
REG DELETE "HKCR\mplayerc64.aac" /f
REG DELETE "HKCR\mplayerc64.ac3" /f
REG DELETE "HKCR\mplayerc64.aif" /f
REG DELETE "HKCR\mplayerc64.aifc" /f
REG DELETE "HKCR\mplayerc64.aiff" /f
REG DELETE "HKCR\mplayerc64.alac" /f
REG DELETE "HKCR\mplayerc64.amr" /f
REG DELETE "HKCR\mplayerc64.amv" /f
REG DELETE "HKCR\mplayerc64.aob" /f
REG DELETE "HKCR\mplayerc64.ape" /f
REG DELETE "HKCR\mplayerc64.apl" /f
REG DELETE "HKCR\mplayerc64.asf" /f
REG DELETE "HKCR\mplayerc64.asx" /f
REG DELETE "HKCR\mplayerc64.au" /f
REG DELETE "HKCR\mplayerc64.avi" /f
REG DELETE "HKCR\mplayerc64.bdmv" /f
REG DELETE "HKCR\mplayerc64.bik" /f
REG DELETE "HKCR\mplayerc64.caf" /f
REG DELETE "HKCR\mplayerc64.cda" /f
REG DELETE "HKCR\mplayerc64.cue" /f
REG DELETE "HKCR\mplayerc64.dav" /f
REG DELETE "HKCR\mplayerc64.divx" /f
REG DELETE "HKCR\mplayerc64.dsa" /f
REG DELETE "HKCR\mplayerc64.dsm" /f
REG DELETE "HKCR\mplayerc64.dss" /f
REG DELETE "HKCR\mplayerc64.dsv" /f
REG DELETE "HKCR\mplayerc64.dts" /f
REG DELETE "HKCR\mplayerc64.dtshd" /f
REG DELETE "HKCR\mplayerc64.dtsma" /f
REG DELETE "HKCR\mplayerc64.dv" /f
REG DELETE "HKCR\mplayerc64.evo" /f
REG DELETE "HKCR\mplayerc64.f4v" /f
REG DELETE "HKCR\mplayerc64.flac" /f
REG DELETE "HKCR\mplayerc64.flc" /f
REG DELETE "HKCR\mplayerc64.fli" /f
REG DELETE "HKCR\mplayerc64.flic" /f
REG DELETE "HKCR\mplayerc64.flv" /f
REG DELETE "HKCR\mplayerc64.hdmov" /f
REG DELETE "HKCR\mplayerc64.ifo" /f
REG DELETE "HKCR\mplayerc64.ivf" /f
REG DELETE "HKCR\mplayerc64.m1a" /f
REG DELETE "HKCR\mplayerc64.m1v" /f
REG DELETE "HKCR\mplayerc64.m2a" /f
REG DELETE "HKCR\mplayerc64.m2p" /f
REG DELETE "HKCR\mplayerc64.m2t" /f
REG DELETE "HKCR\mplayerc64.m2ts" /f
REG DELETE "HKCR\mplayerc64.m2v" /f
REG DELETE "HKCR\mplayerc64.m3u" /f
REG DELETE "HKCR\mplayerc64.m3u8" /f
REG DELETE "HKCR\mplayerc64.m4a" /f
REG DELETE "HKCR\mplayerc64.m4b" /f
REG DELETE "HKCR\mplayerc64.m4r" /f
REG DELETE "HKCR\mplayerc64.m4v" /f
REG DELETE "HKCR\mplayerc64.mid" /f
REG DELETE "HKCR\mplayerc64.midi" /f
REG DELETE "HKCR\mplayerc64.mk3d" /f
REG DELETE "HKCR\mplayerc64.mka" /f
REG DELETE "HKCR\mplayerc64.mkv" /f
REG DELETE "HKCR\mplayerc64.mlp" /f
REG DELETE "HKCR\mplayerc64.mov" /f
REG DELETE "HKCR\mplayerc64.mp2" /f
REG DELETE "HKCR\mplayerc64.mp2v" /f
REG DELETE "HKCR\mplayerc64.mp3" /f
REG DELETE "HKCR\mplayerc64.mp4" /f
REG DELETE "HKCR\mplayerc64.mp4v" /f
REG DELETE "HKCR\mplayerc64.mpa" /f
REG DELETE "HKCR\mplayerc64.mpc" /f
REG DELETE "HKCR\mplayerc64.mpcpl" /f
REG DELETE "HKCR\mplayerc64.mpe" /f
REG DELETE "HKCR\mplayerc64.mpeg" /f
REG DELETE "HKCR\mplayerc64.mpg" /f
REG DELETE "HKCR\mplayerc64.mpl" /f
REG DELETE "HKCR\mplayerc64.mpls" /f
REG DELETE "HKCR\mplayerc64.mpv2" /f
REG DELETE "HKCR\mplayerc64.mpv4" /f
REG DELETE "HKCR\mplayerc64.mts" /f
REG DELETE "HKCR\mplayerc64.mxf" /f
REG DELETE "HKCR\mplayerc64.ofr" /f
REG DELETE "HKCR\mplayerc64.ofs" /f
REG DELETE "HKCR\mplayerc64.oga" /f
REG DELETE "HKCR\mplayerc64.ogg" /f
REG DELETE "HKCR\mplayerc64.ogm" /f
REG DELETE "HKCR\mplayerc64.ogv" /f
REG DELETE "HKCR\mplayerc64.opus" /f
REG DELETE "HKCR\mplayerc64.pls" /f
REG DELETE "HKCR\mplayerc64.pva" /f
REG DELETE "HKCR\mplayerc64.ra" /f
REG DELETE "HKCR\mplayerc64.ram" /f
REG DELETE "HKCR\mplayerc64.rec" /f
REG DELETE "HKCR\mplayerc64.rm" /f
REG DELETE "HKCR\mplayerc64.rmi" /f
REG DELETE "HKCR\mplayerc64.rmm" /f
REG DELETE "HKCR\mplayerc64.rmvb" /f
REG DELETE "HKCR\mplayerc64.rp" /f
REG DELETE "HKCR\mplayerc64.rpm" /f
REG DELETE "HKCR\mplayerc64.rt" /f
REG DELETE "HKCR\mplayerc64.smi" /f
REG DELETE "HKCR\mplayerc64.smil" /f
REG DELETE "HKCR\mplayerc64.smk" /f
REG DELETE "HKCR\mplayerc64.snd" /f
REG DELETE "HKCR\mplayerc64.spx" /f
REG DELETE "HKCR\mplayerc64.ssif" /f
REG DELETE "HKCR\mplayerc64.swf" /f
REG DELETE "HKCR\mplayerc64.tak" /f
REG DELETE "HKCR\mplayerc64.thd" /f
REG DELETE "HKCR\mplayerc64.tp" /f
REG DELETE "HKCR\mplayerc64.trp" /f
REG DELETE "HKCR\mplayerc64.ts" /f
REG DELETE "HKCR\mplayerc64.tta" /f
REG DELETE "HKCR\mplayerc64.vob" /f
REG DELETE "HKCR\mplayerc64.wav" /f
REG DELETE "HKCR\mplayerc64.wax" /f
REG DELETE "HKCR\mplayerc64.weba" /f
REG DELETE "HKCR\mplayerc64.webm" /f
REG DELETE "HKCR\mplayerc64.wm" /f
REG DELETE "HKCR\mplayerc64.wma" /f
REG DELETE "HKCR\mplayerc64.wmp" /f
REG DELETE "HKCR\mplayerc64.wmv" /f
REG DELETE "HKCR\mplayerc64.wmx" /f
REG DELETE "HKCR\mplayerc64.wv" /f
REG DELETE "HKCR\mplayerc64.wvx" /f

**Comment by clsid2** [comment]
The /unregall command line parameter for the player removes those keys.

**Comment by olympus123** [comment]
> The /unregall command line parameter for the player removes those keys.

Yes, but the uninstaller when it uninstalls/deletes mpc-hc64.exe it should call mpc-hc64.exe /unregall before deleting it to make sure no leftovers of it are left in the system.
If it doesn't and user doesn't know about mpc commandline options and doesn't run unregall before uninstalling mpc-hc then he ends up with broken registrations in his system.
I have experience with inno and it can be done with a pascal code call.
Is the inno iss available publically to have a look at it?

**Comment by ale5000-git** [comment]
> Is the inno iss available publically to have a look at it?

https://github.com/clsid2/mpc-hc/tree/develop/distrib

**Comment by olympus123** [comment]
> > Is the inno iss available publically to have a look at it?
> 
> https://github.com/clsid2/mpc-hc/tree/develop/distrib

Thanks I will look at it.

**Comment by olympus123** [comment]
> > Is the inno iss available publically to have a look at it?
> 
> https://github.com/clsid2/mpc-hc/tree/develop/distrib

It's easier than I thought to be added, the iss file is very well done, so no pascal calls needed to find the location of mpc-hc exe.
A simple section added would do it.

[UninstallRun]
Filename: "{app}\{#mpchc_exe}"; Parameters: "/unregall"; WorkingDir: {app}; Flags: skipifdoesntexist waituntilterminated

For making it optional It could be added in the CleanUpSettingsAndFiles procedure, but I don't believe that's the place of it, this is a nessecary clean up of the registry, it shoudn't be an option in my opinion.
CleanUpSettingsAndFiles procedure with it



procedure CleanUpSettingsAndFiles();
var
    ExecCode: Integer;
begin
  Exec(ExpandConstant('{app}\{#mpchc_exe}'), '/unregall', '', SW_HIDE, ewWaitUntilTerminated, AppExecCode);
  DeleteFile(ExpandConstant('{app}\{#mpchc_ini}'));
  DelTree(ExpandConstant('{userappdata}\MPC-HC\ShaderCache'), True, True, True);  
  DeleteFile(ExpandConstant('{userappdata}\MPC-HC\default.mpcpl'));
  RemoveDir(ExpandConstant('{userappdata}\MPC-HC'));
  RegDeleteKeyIncludingSubkeys(HKCU, 'Software\MPC-HC\Filters');
  RegDeleteKeyIncludingSubkeys(HKCU, 'Software\MPC-HC\MPC-HC');
  RegDeleteKeyIfEmpty(HKCU, 'Software\MPC-HC');
end;

**Comment by adipose** [comment]
Submit a patch.

**Comment by clsid2** [comment]
Already done.
